src/cascadia/TerminalSettingsModel/FileUtils.cpp

```diff
@@ -123,7 +123,11 @@ namespace Microsoft::Terminal::Settings::Model
 
     void WriteUTF8FileAtomic(const std::filesystem::path& path, const std::string_view content)
     {
-        auto tmpPath = path;
+        // GH#10787: In the case of symbolic link, ensure that we modify the target rather than the link itself.
+        // We append the paths manually rather than using "canonical" method to support scenario in which link target doesn't exist
+        const auto resolvedPath = std::filesystem::is_symlink(path) ? path.parent_path() / std::filesystem::read_symlink(path) : path;

```

Comment:
What if the target of the symlink is another symlink? Consider using [`std::filesystem::canonical()`](https://docs.microsoft.com/en-us/cpp/standard-library/filesystem-functions?view=msvc-160#canonical), which is implemented in Windows via `GetFinalPathNameByHandleW()`.

Reply:
Yeah. This is how I started. The problem with canonical is that it requires the target to exist. I am not sure which is more a corner case: a link to a link or link to non- existing file.

Reply:
You could handle the exception / error by manually resolving the target path.

Reply:
> You could handle the exception / error by manually resolving the target path.

So if there is a link to a link to non-existing file, the canonical will throw / return error. And in this case we will need to resolve path manually, by appending within a loop. In this case the question is why not to resolve manually in a loop in the first place. 

So, IMHO, the options are:
1. resolve manually by replacing if with while
2. use canonical, with disclaimer the path needs to exist

WDYT?

Reply:
It seems to me like we should be redundant enough to be able to handle both smylinks that point to symlinks and paths that don't exist. Like, what about when the file is first written? That's a case where the path won't exist yet.

Reply:
So I continued with the first option of manual resolution in while loop. Tested the following paths:
* no file
* file
* a link to a file
* a link to a link to a file
* a link to a non-existing file